File: frontend-bk/src/services/api/auth.service.ts
import type { ApiResponse } from '@/interfaces/api.interface'
import type { LoginRequest, LoginResponse } from '@/interfaces/auth.interface'
import { BaseService } from './base.service'

export class AuthService extends BaseService {
  constructor() {
    super('/api/auth')
  }

  async login(credentials: LoginRequest): Promise<ApiResponse<LoginResponse>> {
    return this.post<LoginResponse>('/login', credentials)
  }

  async refreshToken(refreshToken: string): Promise<ApiResponse<LoginResponse>> {
    return this.post<LoginResponse>('/refresh-token', { refreshToken })
  }

  async logout(): Promise<ApiResponse<void>> {
    return this.post<void>('/logout')
  }
}

export const authService = new AuthService()

File: backend/domain/auth-domain/src/main/java/org/atlas/domain/auth/service/AuthService.java
package org.atlas.domain.auth.service;

import org.atlas.domain.auth.model.GenerateOneTimeTokenRequest;
import org.atlas.domain.auth.model.GenerateOneTimeTokenResponse;
import org.atlas.domain.auth.model.LoginRequest;
import org.atlas.domain.auth.model.LoginResponse;
import org.atlas.domain.auth.model.OneTimeTokenLoginRequest;
import org.atlas.domain.auth.model.RefreshTokenRequest;
import org.atlas.domain.auth.model.RefreshTokenResponse;

public interface AuthService {

  LoginResponse login(LoginRequest request) throws Exception;

  LoginResponse oneTimeTokenLogin(OneTimeTokenLoginRequest request) throws Exception;

  GenerateOneTimeTokenResponse generateOneTimeToken(GenerateOneTimeTokenRequest request);

  RefreshTokenResponse refreshToken(RefreshTokenRequest request) throws Exception;

  void logout();
}
